mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -602,6 +656,12 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
         return this
     }
 
+    private fun <S : MvRxState> assertSubscribeToDifferentViewModel(viewModel: BaseMvRxViewModel<S>) {
+        if (this == viewModel) {
```

Comment:
could also do
```
require(this != viewModel) {
 "This method is for subscribing to other view models. Please pass a different instance as the argument."
}
```

Still throws `IllegalArgumentException` and is a little shorter and more idiomatic kotlin.